### DISCUSSION TITLE
기초 프로젝트 3주차 팀회고(최종)

### DISCUSSION THREAD

**Discussion by sulyoungoh** [codeit-bootcamp-frontend/3-Weekly-Mission#246] [회고]
기초 프로젝트 무사히 마무리 하신 여러분,
너무 너무 고생 많으셨어요💜
첫 협업 프로젝트였던 만큼 많이 고생하신 걸 알기 때문에, 어제 보여주셨던 결과물이 너무나 소중하고 값지게 느껴지네요🥰
고생하신 만큼 이번주 주말만큼은 푹 쉬시면서 행복한 시간 보내시길 바랍니다!

2주간의 팀 프로젝트를 돌아보며 서로 고마웠던 점이 있다면 솔직하게 나누어보고, 속상했던 것이 있다면 함께 풀 수 있는 시간이 되었으면 좋겠어요.
저번과 같이 KPT 회고로 진행해주신 후, __**이번주 일요일(02/04)까지**__ 댓글 남겨주세요.

**Comment by ohddang** [comment]
# 3주차 (1.29 ~ 2.04)

## 🌼keep (현재 우리 팀이 잘 하고 있어서 계속 유지하면 좋을 것 같은 부분)

- 크게 부딪히는 일 없이 각자 할 일 묵묵히 한 게 좋았다.
- 적용한 기술의 사용 이유와 목표를 잘 정리해주셔서 참고하면서 함께 공부하기 좋았다.
- 다들 코딩에 진심이라 같이 진심이 될 수 있었다.
- 공부를 많이 할 수 있는 환경을 만들어주셔서 좋았다.
- 서로 격려를 해줘서 좋았다.

## 🛸**Problem** (현재 우리 팀이 가지고 있는 문제)

- 조금 더 아이디어를 가지고 더 적극적으로 바꿔보려는 노력을 하면 좋았을 것 같다.
- Jira에 이슈카드를 바로바로 등록하고 가져가서 하지 못했다.
- 컴포넌트 재사용성을 좀 더 고려해서 컴포넌트의 순수성을 유지했으면 좋았을것 같다.
- 디테일한 부분이 부족했던거 같다.

## ✨try (Problem에 적은 것을 해결하기 위해 우리 팀이 다음에 시도 해볼 만한 것, 혹은 Keep 중에서도 더 개선하거나 시도해보고 싶은 것)

- 다양한 시도를 해보려고 노력했지만 다음엔 조금 더 열심히 시도해도 좋을 것 같다.
- 개발을 할 때 확장성과 실제 서비스 환경을 고려하여 개발하면 좋을거 같다.
- 디자인을 좀 더 신경썼으면 좋았을 것 같다.
- 코드를 보기 좋게 정리하면 좋을 것 같다.

**Comment by DanteSnow** [comment]
## 5팀


### 💡**Keep**

- 모두가 밤을 새서라도 마감 기한을 지키려고 프로젝트에 열정적으로 임해서 더욱 열심히 하게 되는 원동력이 되었다.
- 기능 구현에 문제가 생겼을 때 모두 자기 일 처럼 도와주려고 노력했고, 덕분에 대부분의 공정이 순조롭게 흘러갔다.
- 처음보는 기능들을 구현하는 것에서 끝내는게 아니라 계속해서 서로 공유해서 배울점이 정말 많았다.
- 팀 프로젝트라는 것에 대한 막연한 심리적 거부감이 있었는데 막상 해보고 나니 어느정도 해소되었다.
- 강의에서 영상으로 보는것과 내가 실제로 그 기능을 써서 무언가를 만들어보는것은 학습의 흡수율의 차원이 달랐다.
- 팀원들이 모두 나보다 실력자들이라 코딩에 관한 지식은 물론, 이번 교육과정에 대한 애티튜드 및 개인적인 스케줄 관리등 많은것을 보고 배울 수 있었다.
- 팀프로젝트를 하면서 협업을 어떻게 하는지 배울 수 있었다.
- 깃을 사용할 때 항상 불안함과 두려운이 있었는데 이번 프로젝트를 통해 조금 개선이되었다.
- 새로운 기능을 구현해볼 수 있어서 새로운 지식을 얻게 되었다.
- 팀원 모두 일정을 잘 맞춰 주어서 좋았다.
- 큰 틀을 잘 정해둔 덕에 수월하게 진행된 것 같다.
- 최대한 깃허브를 이용해서 작업한 부분이 좋았다. 덕분에 깃허브에 대한 이해도가 올라간 것 같다.
- 이번 프로젝트에서 여유를 가지면서 코드 작업을 했던 부분이 좋았다. 해결이 안되는 문제가 생겼을 때, 컴퓨터 앞에 앉아서 계속 코드를 작업하는 것 보다는 잠깐 휴식을 취하면서 머릿속으로 어떻게 코드를 구현할지 생각을 정리한 것이 문제 해결에 큰 도움이 되었다.

### ❓**Problem**

- 한 페이지에 여러 사람이 붙어 기능 개발을 하면서 생긴 문제가 있었다. 팀원들 각자의 코드를 작성하는 방식이 달라서 코드를 이해하는데 오래 걸렸고, 추가 기능을 구현 할 때와 버그 픽스를 할 때 등 유지 보수에 매우 큰 어려움을 겪었다.
- 이름이 기초프로젝트인데 나한테는 이번 프로젝트가 마냥 쉽지는 않았다. 그렇다면 다음 프로젝트때는 또 어떤 시련들이 나를 기다리고 있을까?
- 초반에 당장 할 수 있는 것들은 스케줄에 상관없이 다 몰아서 해버렸는데, 그러고 나니 후반에는 힘이 빠져서 이제 그만 하고싶다는 못된 감정들이 생겼다.
- 잘 하고 싶은데 머리가 안 따라와주니까 딱밤을 좀 때리고 싶었다.
- 처음부터 공용 컴포넌트를 설계해서 진행했다면 더 좋았을 것 같다.
- 처음에는 개발 회고를 매일 작성하려고 했는데 막상 하다보니 정신없이 코드만 짜고, 그 과정에서 겪은 문제점들이나 해결 방법을 기록하지 못했다.
- 스스로 며칠 안에는 해결해야지~ 했던 결심이 해이해지고, 프로젝트 일정에만 따라갔던 것 같다. 조금 더 미리미리 작업할 수 있었을텐데.
- 코드 정리를 많이 못한게 매우 아쉽다.
- 어떤 의견이 나랑 다른 의견이라고 생각이 들었던 적이 있었는데, 반대되는 이유에 대해 제대로 생각하지 못해서 말을 못한적이 몇 번 있었다.
- 페이지 단위로 작업을 하지 못해서 생긴 이슈가 있었다. 한 페이지에서 같이 작업하다가 PR이 올라오면 내가 작성한 코드들도 다시 수정해야한다. 추가적인 기능을 구현할 때도, 다른 팀원이 작성한 코드를 이해해야 했기 때문에 시간이 많이 걸렸다.

### 🏃**Try**

- 이런 방식의 설계는 리팩토링은 필수로 진행을 해주거나, 리팩토링이 필요없을 정도의 컨벤션을 확실하게 설계하고 갔어야 했다고 생각한다.
- 한 페이지를 한 사람이 맡는 방식의 프로젝트 진행 방식으로 진행해봐도 좋을 것 같다는 생각을 했다.
- 상대적으로 쉬운 난이도의 페이지가 있다면 해당 팀원이 빨리 끝내고 다른 팀원들을 도와주었다면 더 순조롭게 진행이 가능했을 것 같다.

**Comment by soyxxn** [comment]
### **2팀**

**🩷Keep**
- 본인이 맡은 일을 포기하지 않고 끝까지 책임지려는 모습 !
- 맡은 일을 다 완수했음에도 더욱 꼼꼼하게 보완하고, 발전시켜 나가려는 모습 !
- 서로를 배려하면서도 보완해야 할 부분은 집어서 피드백 해주는 점 !
- 막혔을 때 자기 일처럼 나서서 해결해주려는 모습 !
- 일 때문에 토요일 팀미팅이 늦어졌는데 배려해주셔서 너무 감사했다 ! (너무x10000)

**🤔Problem**
- 공통 컴포넌트를 미리 분리하지 않은 점
- 프로젝트 일정 관리를 세세하게 짜놓지 않아 후반부에 갑자기 추가하거나 수정해야 할 부분이 늘어난 점
- 맞닥뜨린 오류나 코딩 과정을 체계적으로 정리하지 않아 다시 봤을 때 기억이 잘 안 났던 점
- 발표가 끝나고 해이해져서 리팩토링을 못한 점
- 초반에 어떻게 해야할 지 막막함에 시간을 많이 허비했던 점

**💡Try**
- 재사용성을 높이기 위해 공통 컴포넌트와 구조화, 추상화에 좀 더 관심을 가지기 !
- 프로젝트 일정 관리를 좀 더 체계적이고 촘촘하게 하기 !
- 코드 리팩토링과, 그 과정을 정리하며 진행하면 나중에 봤을 때 도움이 더 될 것 같다 !
- 초반의 막막함을 떨쳐내고 시작을 빨리 해보기 !
- 위와 같이 이번 프로젝트를 통해 배운 것들을 다음에는 여러 방향으로 적용해보고 싶다 !

-----------------------------------------------------------------------------------------------------------

2팀 최고 ! 다들 너무 너무 수고 많으셨고 감사했습니다 !
함께 프로젝트를 진행하게 돼서 너무 행복했어요 ! 🥰🩷🩷🩷 - 괴도

**Comment by yuyoni** [comment]
# 1팀 💜

## 👍 Keep

- 유일한 rolling 프로젝트 진행! 시작하면서 완성 할 수 있을지, 낯선 팀프로젝트 잘 진행할 수 있을지 많은 걱정이 있으셨을텐데 한 명도 빠짐없이 모두 성실히, 끝까지 잘 해내신 것 같습니다. 시작과 끝을 비교해보면 팀 으로보나 개개인으로보나 많은 발전이 보이는 것 같습니다 : ) 고생하셨습니다!
- 다들 과제도 바쁘셨던 걸로 알고있는데, 각자에게 주어진 역할을 큰 갈등없이 기한 내에 완수했다는 것 만으로도 기초프로젝트로서 성공한 것 같습니다. 개인적으로는 성격이 급하다고 생각해 다른분들이 느끼기에는 재촉하는 것 같을까봐 걱정했는데 다들 덕분에 잘 마쳤다고 말씀해주셔서 감사했습니다. 서로 칭찬해주는 부분 너무 좋았습니다!!
- 처음에 걱정 많으셨지만 열심히 잘 따라와준 선영님!
소리 소문없이(?) 하셔서 약간 걱정이 들긴 했지만 멋지게 일정을 잘 맞춰주신 율민님!
많은 기능들을 담당하느라 애 많이 쓰신 인덕님(특히 아프신데도 불구하고 끝까지 기능 다 구현하신거 정말 섹시했고, 도움이 되지 못해 죄송한 마음이 많이 듭니다 ㅜㅜ) 
처음부터 끝까지 성실하고 적극적으로 임해주신 재명님!
그리고 마지막 외쳐 갓 ! 유 ! 연 ! 팀장으로 책임감도 강하고, 잘 이끌어주시려고 하고 다들 하기 싫어하는 발표나 정리 부분들 도맡아 해주셔서 진짜 감사하게 생각하고 있습니다.
프로젝트 진행 하는 동안 지칠 때도 있었을텐데 모두 파이팅 넘치는 모습이 보기 좋았고, 서로를 응원해주는 마음들이 잘 느껴져 프로젝트 기간동안 좋은 기운을 많이 얻은 것 같습니다.
다들 고생 많으셨습니다 !
- 프로젝트를 시작하게 되면 팀운이 정말 모든 것을 결정한다고들 하는데 정말 행운아였던 것 같습니다! 프로젝트를 진행하면서 한번도 혼자 외롭게 개발한다는 느낌을 받지 않고 같이 개발한다는 느낌을 받을 수 있어서 팀원분들한테 정말 감사한게 많습니다. 프로젝트를 진행하면서 개인적으로 많은 것을 스스로 공부해보고 부딪혀보며 개발 역량을 성장시킨 것도 좋았지만, 무엇보다 팀이라는 단체에서 팀원들과 어떻게 소통하고 협업할 수 있는지에 대해 배울 수 있어서 좋았던 것 같습니다. 첫 프로젝트였던만큼 처음에는 걱정과 부담감이 있었지만, 끝에는 뿌듯함과 약간의 아쉬움이 남았던 것 같습니다. 다들 옆에서 잘 도와주셔서 감사했습니다~ 모두 수고 많으셨습니다~!
- 서로 피드백도 많이하고 코드를 공유하면서 개발자로써 많이 성장할수있었던거같습니다. 고생하셨습니다!

## 💭 Problem

- 전반적으로 소통이 조금 부족했다고 느낍니다. 문제가 발생하면 즉시 공유하고 같이 해결해나가는 과정인데 내가 폐가 되지 않을까, 혹은 혼자 해결해야하지 않을까 하는 두려움들이 방해하는 부분이 있으셨을거라고 생각해요. 이건 아마 한국인 특이 아닐까요?ㅎㅎ 하지만 그런 부분들이 도리어 빠른 해결이나 진척을 가로막는 부분들로 작용한 부분들도 있었던 것으로 생각합니다.
- 새로운 시도에 대해 두려움이 좀 있는 것 같다고 느꼈습니다. 모르는 개념이라도 사용해보면서 배울 수 있는데 아는 개념에만 갇혀서 진행한 느낌이었습니다.
- 모두가 적극적으로 의견을 피력하지는 않았다고 생각합니다. 큰 갈등이나 대립이 없어서 다행이라고 볼 수도 있지만 개인적으로는 어떤 것이 좋은 의견인지 비교해보는 과정 없이 그냥 한쪽만의 의견으로 진행되는일이 대립보다 더 좋지 않은 것 같다고 생각합니다.
- 문제가 생겼을 때 함께 해결해보자라는 느낌보단 각자의 업무처럼 느껴졌습니다. 물론 오류 발생시 해당 기능 담당자가 해결하는게 맞다고 생각하긴 하지만 아직 공부하는 입장에서 또 팀 협업을 경험하는게 주 목적인 만큼 함께 의견을 주고 받으며 해결해 나가는 것도 좋았을 것 같단 생각이 듭니다.
- 첫 프로젝트였던 만큼 부딪혀보려는 시도가 많이 없었던 것 같습니다. 본인의 실수로 프로젝트에 차질이 생길것 같은 마음때문에 큰 도전을 못했던 경우도 있었던 것 같습니다. 사실 배우는 과정이고 이러한 부딪혀 보는 과정에서 더 많은 것을 얻을 수 있을 것 같습니다.

## ⭐️ Try

- 팀은 여러분을 혼내거나 해치지 않아요~다음 프로젝트에는 문제도 가감없이 공유하고, 자신의 의견도 자신있게 피력할 수 있는 동료로 만나는 날을 기대 해봅니다 : )
- 다음번에는 새로운 기술을 적용해보고, 새로운 툴을 사용해보는 프로젝트 경험을 해보는 것, 그리고 제로부터 다 같이 세팅해보는 것이 프로젝트 경험면에서는 더 좋을 것 같습니다!
- 미리 얘기할 주제에 대해 찾아보고 어떤 게 더 좋은 것 같은지 의견을 나눌 수 있을 정도로 본인의 생각을 확고하게! 정하면 훨씬 질 좋은 회의가 될 수 있을 것 같습니다ㅎㅎ
- 아직 공부하는 입장이니 모르는게 있는건 당연하다 생각합니다. 혹시 내가 틀린건 아닐까? 하는 생각에 발언을 주저하지 말고 의견을 제시해보면 좋을 것 같습니다. 틀린 의견에 대한 피드백도 공부가 될테니까요!
- 아직 배우는 입장이라는 것을 계속 인지했으면 좋겠습니다. 가장 크게 성장할 수 있는 것이 실수로부터 알아갈 때라고 생각하고 있습니다. 또한 팀원분들도 실수했다가 절대 뭐라할 팀이 아니었기에 더 도전을 못했던 것이 아쉬웠던 것 같습니다. 팀원분들도 실수가 발생했을 때 같이 도와줄 것이기 때문에 한번쯤은 부딪혀보는 것도 좋다고 생각합니다.
- 프로젝트는 끝났지만 타입스크립트를 배우기 시작했으니 타입스크립트를 활용해서 코드를 리팩토링 해보는것도 좋을거같습니다.

**Comment by SiWooJinSeok** [comment]
# 6팀

# 👍🏻 Keep
- 자신이 맡은 부분을 기한 내에 마무리하여 PR을 올리고, 코드 리뷰와 코드 수정을 꼼꼼히 했다.

- 모두 최선을 다해 각자의 역할에 맡게 잘 해냈다.

- 팀원이 능동적으로 일거리를 찾아서 척척 다 해결해서 멋있다 !

- 새로운 기술에 대한 두려움이 없다.

# 🤔 Problem
- 로딩, 에러 처리를 못했다는 것. 로딩중 이거나 에러가 났을 때 임시로 보여줄 컴포넌트를 만들지 못해서 아쉽다.

- 스토리북이 익숙하지 않아 만드는 데 어려움이 있었다.

- 추가 기능을 더 넣었으면 좋았을 것 같다.

# 🔥 Try
- 발표가 끝이나면 모두에게 칭찬해주기!

- 프론트엔드 마스터가 되자!!

**Comment by 2zzzyoung** [comment]
# 3팀🌝

### Keep
- 기록된 것을 찾아보고 활용하는 것이 생각보다 중요해 보인다. 모르는 것을 구글링하는 것 뿐만 아니라 직접 작성한 코드를 다시 찾아보며 코드를 재사용함으로서 조금 더 효율적인 코딩이 가능했던 것 같다.
- 깃 충돌을 많이 겪으면서 충돌에 대한 두려움이 많이 사라졌다.
- 프로젝트를 진행하면서 친하게 지내며 좋은 분위기를 만들어서 마음이 편했다.
- 프로젝트가 끝난 뒤에 바로 리팩토링 시간을 가지는 것이 좋았다.

### Problem
- 해야할 작업을 놓치게 되었다. 사소하다고 생각 한 작업을 기록해 두지 않아 우선순위에 밀려 잊게 되었다. 해야할 작업을 꼼꼼하게 기록해 두는 것이 필요해 보인다.
- 생각보다 노션과 깃허브를 많이 활용하지 못 한 것 같고 서로의 코드를 확인할 시간이 없었던 것 같다.
- 서로 내적 친밀감이 많이 쌓여 해야 할 일을 하기 싫어하게 되었던 것 같다.
- 프로젝트를 하면서 마주쳤던 이슈나 문제들을 그냥 넘기지 않고 좀 더 체계적이고 구체적으로 정리하고 기록으로 남겨두면 더 좋을 것 같다.

### Try
- 해야할 작업들을 기록해 두고, 우선순위에 따라 스케줄링을 하면, 보다 효율적으로 코딩을 하여 작업 시간을 단축하고 완성도를 높일 수 있을 거 같다.
- 정해진 기한을 지키기 위해 현재 진행 상황을 더 공유하며 서로 도울 수 있는 부분들은 도와가며 하면 좋을 것 같다.
- 공사 구분을 좀 더 명확하게 하고 평소에 좀 더 주의하면 좋을 것 같다.
- 프로젝트 동안의 이슈와 에러들을 정리해서 노션과 블로그에 정리해볼 예정이다.
